### PR TITLE
Fix emission bugs

### PR DESCRIPTION
**Pull request by hazemakhalek**
## Changes proposed in this Pull Request
This PR fixes a few bugs that occurred in counting the emissions. Main sectors affected are "industry" (oil and coal) as well as the custom sectors "services" and "residential"

## Checklist

- [x] I tested my contribution locally and it seems to work fine.
- [x] Code and workflow changes are sufficiently documented.
- [x] Newly introduced dependencies are added to `envs/environment.yaml` and `envs/environment.docs.yaml`.
- [x] Changes in configuration options are added in all of `config.default.yaml`, `config.tutorial.yaml`, and `test/config.test1.yaml`.
- [ ] Changes in configuration options are also documented in `doc/configtables/*.csv` and line references are adjusted in `doc/configuration.rst` and `doc/tutorial.rst`.
- [ ] A note for the release notes `doc/release_notes.rst` is amended in the format of previous release notes, including reference to the requested PR.

### CODE REVIEW

File: scripts/prepare_sector_network.py [hunk 1]
```diff
@@ -1175,7 +1175,7 @@ def add_industry(n, costs):
         n.loads.loc[nodes + co2_release, "p_set"].sum()
         * costs.at["oil", "CO2 intensity"]
         # - industrial_demand["process emission from feedstock"].sum()
-        / 8760
+        # / 8760
```
Comment (energyLS) [inline scripts/prepare_sector_network.py:1178]: Why is the 8760 division kicked out? To me it seems like `n.loads.loc[nodes + co2_release, "p_set"].sum()
        * costs.at["oil", "CO2 intensity"]` gives an annual value, hence it should be divided by 8760 if I am not wrong here. But probably the sum is over all nodes and not over the time dimension, hence kicking out the 8760 is a good bug catch

File: scripts/prepare_sector_network.py [hunk 2]
```diff
@@ -1878,15 +1893,61 @@ def add_services(n, costs):
         carrier="services biomass",
         p_set=p_set_biomass,
     )
-    co2 = (
```
Comment (energyLS) [inline scripts/prepare_sector_network.py:1881]: why is this kicked out?

File: scripts/prepare_sector_network.py [hunk 3]
```diff
@@ -461,7 +461,7 @@ def add_biomass(n, costs):
         nodes + " biomass EOP",
         bus0=spatial.biomass.nodes,
         bus1=nodes,
-        bus2="co2 atmosphere",
+        # bus2="co2 atmosphere",
```
Comment (energyLS) [inline scripts/prepare_sector_network.py:464]: why is this kicked out?

File: scripts/prepare_sector_network.py [hunk 4]
```diff
@@ -470,7 +470,7 @@ def add_biomass(n, costs):
         p_nom_extendable=True,
         carrier=biomass_gen,
         efficiency=costs.at[biomass_gen, "efficiency"],
-        efficiency2=costs.at["solid biomass", "CO2 intensity"],
+        # efficiency2=costs.at["solid biomass", "CO2 intensity"],
```
Comment (energyLS) [inline scripts/prepare_sector_network.py:473]: why is this kicked out?

**Review by energyLS — COMMENTED** [review]
It seems like you have kicked out co2 emissions from biogas/biomass. I think it makes sense if we don't account for negative emissions when introducing biogas/biomass in `spatial.biomass.nodes`, which I am not quite sure about

File: scripts/prepare_sector_network.py (hunk 3 above)
Comment (hazemakhalek) [reply scripts/prepare_sector_network.py:464]: upon further research, I think we should not count the emissions from biomass as long as we set a "sustainable" biomass potential values. It is assumed that biomass emissions will be reabsorbed by forests/trees planted to replace those cut down to burn.

Comment (hazemakhalek) [reply scripts/prepare_sector_network.py:464]: This is also the same approach done in PyPSA-Eur

File: scripts/prepare_sector_network.py (hunk 4 above)
Comment (hazemakhalek) [reply scripts/prepare_sector_network.py:473]: See the previous comment :)

File: scripts/prepare_sector_network.py (hunk 2 above)
Comment (hazemakhalek) [reply scripts/prepare_sector_network.py:1881]: upon further research, I think we should not count the emissions from biomass as long as we set a "sustainable" biomass potential values. It is assumed that biomass emissions will be reabsorbed by forests/trees planted to replace those cut down to burn.

Comment (hazemakhalek) [reply scripts/prepare_sector_network.py:1881]: This is also the same approach done in PyPSA-Eur

Comment (hazemakhalek) [reply scripts/prepare_sector_network.py:1881]: We can set it as a config parameter and set its default value to false

Comment (energyLS) [reply scripts/prepare_sector_network.py:1881]: > upon further research, I think we should not count the emissions from biomass as long as we set a "sustainable" biomass potential values. It is assumed that biomass emissions will be reabsorbed by forests/trees planted to replace those cut down to burn.

I totally agree here.

**Review by energyLS — APPROVED** [review]
All issues resolved

**Pull request merged**